seatsio-android-component/src/main/java/io/seats/eventManager/StaticModeConfig.java

```diff
@@ -2,18 +2,75 @@
 
 import com.google.gson.annotations.Expose;
 
-import java.util.Map;
+import java.util.List;
 import java.util.function.Function;
 
 import io.seats.seatingChart.SeatsioObject;
-import io.seats.utils.Function3;
+import io.seats.utils.Either;
 
-public class StaticModeConfig extends EventManagerConfig {
+public class StaticModeConfig extends EventManagerConfig implements HasTooltipContents {
 
     @Expose
     public ObjectPopover objectPopover;
 
-    public Function<SeatsioObject, String> tooltipContents;
+    public Either<String, Function<SeatsioObject, String>> tooltipContents;
 
-    public Function3<SeatsioObject, String, Map<String, ?>, String> objectIcon;
+    public String objectIconJavascriptFunction;
+
+    public StaticModeConfig() {
+        setMode(EventManagerMode.STATIC);
+    }
+
+    @Override
+    public StaticModeConfig setMode(EventManagerMode mode) {
+        if (mode != EventManagerMode.STATIC) {
+            throw new IllegalArgumentException("Mode must be 'static'");
+        }
+        return this;
+    }
+
+    public void setObjectPopover(ObjectPopover objectPopover) {
+        this.objectPopover = objectPopover;
+    }
+
+    public StaticModeConfig setTooltipContents(Function<SeatsioObject, String> tooltipContents) {
+        this.tooltipContents = Either.right(tooltipContents);
+        return this;
+    }
+
+    public StaticModeConfig setTooltipContents(String tooltipContents) {
```

Comment:
We don't need to support a string for this callback. It's executed outside the iFrame, so a pure Java function is fine.